apps/microtvm/zephyr/template_project/microtvm_api_server.py

```diff
@@ -262,7 +262,11 @@ def _get_openocd_device_args(serial_number: str = None):
     return ["--serial", generic_find_serial_port(serial_number)]
 
 
-def _get_nrf_device_args(serial_number: str = None):
+def _get_nrf_device_args(serial_number: str = None) -> list:
+    # iSerial has string type which could mistmatch with
+    # the output of `nrfjprog --ids`. Example: 001050007848 vs 1050007848
+    serial_number = str(int(serial_number))
```

Comment:
Would it be cleaner to do this? Not sure.
```suggestion
    serial_number = serial_number.lstrip("0")
```

Reply:
done